doc/mgmt/redis_client_manager.md

```diff
@@ -0,0 +1,176 @@
+- curTransactionalClients: The number of active transactional clients that were provided by RCM.
+- totalPoolClientsRequested: The number of pool clients that were requested through RCM.
+- totalTransactionalClientsRequested: The number of transactional clients that were requested through RCM.
+
+Additionally go-redis has internal counters for Connection Pools that can be accessed for each Redis Client in the cache. More details are provided [here](https://pkg.go.dev/github.com/go-redis/redis/v8@v8.11.5/internal/pool#Stats).
+
+The counters can be fetched through another function provided by RCM, called RedisClientManagerCounters().
+
+#### Command Line Flag
+
+The connection pools feature can be disabled through a new boolean command line flag: "use_connection_pools"
+
+This flag is set to true by default, enabling connection pools.
+
+When "use_connection_pools" is false, all calls to RedisClient() will just call TransactionalRedisClient(). This will return a unique Redis Client to each caller and bypass the pool feature completely.
+
+#### Refactoring Callsites
+
+In all places where a Redis Client is not used to perform a transactional operation, the call will be refactored to use RedisClient(). This will result in the caller now using a shared Redis Client with a connection pool. This will significantly reduce the number of new Redis Clients and connections to Redis.
+
+In all places where a transactional operation is performed, the call will be refactored to use TransactionalRedisClient(). This will give the caller a new Redis Client that is unique. If more options need to be specified, the call will be refactored to use TransactionalRedisClientWithOpts().
+
+All calls to redis.Close() will be replaced with calls to CloseRedisClient(). This will ensure that Redis Clients that use connection pools are not closed.
+
+### SAI API
+
+This HLD does not include any changes to the SAI API.
+
+### Configuration and Management
+Although this change interacts with configuration and management, it should be a transparent change. There is no change in functionality. The purpose of this HLD is to reduce the stress of configuration and management on Redis.
+
+#### CLI/YANG model Enhancements
+
+This HLD does not include any changes to CLI or YANG models.
+
+#### Config DB Enhancements
+
+This HLD does not include any changes to the Config DB.
+
+### Warmboot and Fastboot Design Impact
+
+This HLD does not have any implications on Warmboot/Fastboot.
+
+### Memory Consumption
+
+Increased memory usage from this HLD should be minimal. Increased memory  comes from the 10 (DBNum) Redis pool clients that are created during initialization and stay open for the life of the server. There are also new counters that are being introduced that live in memory.
+
+### Restrictions/Limitations
+
+RCM lives in the translib/db package. This package imports CVL, so refactoring calls in CVL to use RCM would create a circular dependency.
```

Comment:
Thanks for the suggestion. I think it makes the most sense for this code to live in the `sonic-mgmt-common/translib/db` directory because the API will be integrated with the db_redis_opts.go and db_stats.go. We can consider moving it out of this directory if necessary.